File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {
    private LocalDateTime gigasecond;

    Gigasecond(LocalDate birthDate) {
      this(birthDate.atStartOfDay());
    }

    Gigasecond(LocalDateTime birthDateTime) {
      gigasecond = birthDateTime.plusSeconds(1000000000);
    }

    LocalDateTime getDate() {
      return gigasecond;
    }
}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.Month

class Gigasecond(val fromDateTime: LocalDateTime) {
  constructor(fromDate: LocalDate): this(fromDate.atTime(0, 0))
  val date = fromDateTime.plusSeconds(1000000000)
}
